client/src/modules/cohort/index.jsx

```diff
@@ -24,17 +24,23 @@ import GridList from 'common/components/GridList';
 import { ListType } from 'modules/list';
 import MembersBox from 'common/components/Members';
 import { Routes } from 'common/constants/enums';
-import { noOp } from 'common/utils/noOp';
 import CohortHeader from './components/CohortHeader';
+import Preloader from '../../common/components/Preloader';
 
 class Cohort extends PureComponent {
   state = {
-    areArchivedListVisible: false,
+    areArchivedListsVisible: false,
     dialogContext: null,
-    isListPrivate: true
+    isListPrivate: true,
+    pendingForArchivedLists: false,
+    pendingForDetails: false,
+    pendingForListCreation: false,
+    pendingForCohortArchivization: false
   };
 
   componentDidMount() {
+    this.setState({ pendingForDetails: true });
```

Comment:
fetchData already do that inside - so remove this or pull out `pending` update here